packages/gui/src/hooks/useNFT.ts

```diff
@@ -1,34 +1,29 @@
-import { useContext, useState, useCallback, useMemo, useEffect } from 'react';
+import { useContext, useState, useCallback, useEffect, useMemo } from 'react';
 
 import NFTProviderContext from '../components/nfts/provider/NFTProviderContext';
-import getNFTId from '../util/getNFTId';
 
 export default function useNFT(id?: string) {
   const context = useContext(NFTProviderContext);
   if (!context) {
     throw new Error('useNFT must be used within NFTProvider');
   }
 
-  const nftId = useMemo(() => id && getNFTId(id), [id]);
-  const { invalidate, getNft, events } = context;
+  const { invalidate, getNFT, subscribeToNFTChanges } = context;
 
-  const handleInvalidate = useCallback(() => invalidate(nftId), [invalidate, nftId]);
-  const [nftState, setNFTState] = useState(getNft(nftId));
+  const handleInvalidate = useCallback(() => invalidate(id), [invalidate, id]);
+  const [nftState, setNFTState] = useState(() => getNFT(id));
 
-  useEffect(() => {
-    function handleChange(changedNFTId: string, state: ReturnType<typeof getNft>) {
-      if (changedNFTId === nftId) {
-        setNFTState(state);
-      }
-    }
+  useMemo(() => {
+    setNFTState(getNFT(id));
+  }, [id, getNFT]);
 
-    events.on('nftChanged', handleChange);
-    setNFTState(getNft(nftId));
-
-    return () => {
-      events.off('nftChanged', handleChange);
-    };
-  }, [events, getNft, nftId]);
+  useEffect(
+    () =>
+      subscribeToNFTChanges(id, (newNFTState) => {
```

Comment:
Related to my earlier comment, should we at least include a comment here that `subscribeToNFTChanges` returns a cleanup function?